packages/mobile/src/harmony-native/components/Artwork.tsx

```diff
@@ -0,0 +1,147 @@
+import { useEffect, useState } from 'react'
+
+import { useTheme } from '@emotion/react'
+import FastImage from 'react-native-fast-image'
+import Animated, {
+  useAnimatedStyle,
+  useSharedValue,
+  withTiming
+} from 'react-native-reanimated'
+
+import type { FastImageProps } from './FastImage/FastImage'
+import { Skeleton } from './Skeleton'
+import { Box } from './layout/Box/Box'
+import type { BoxProps } from './layout/Box/types'
+import { Flex } from './layout/Flex/Flex'
+
+const AnimatedFastImage = Animated.createAnimatedComponent(FastImage)
+
+export type ArtworkProps = {
+  isLoading?: boolean
+  borderWidth?: number
+  'data-testid'?: string
+  noLoading?: boolean
+} & Partial<Pick<FastImageProps, 'source' | 'priority'>> &
+  BoxProps
+
+/**
+ * The artwork component displays the track cover art and appears in several
+ * locations such as track tiles, track and playlist, pages,
+ * and the sidebar. It can have interactive elements on hover.
+ * This component enhances the listening experience and helps users quickly
+ * identify their favorite tracks.
+ */
+export const Artwork = (props: ArtworkProps) => {
+  const {
+    isLoading: isLoadingProp,
+    source,
+    borderRadius = 's',
+    borderWidth,
+    shadow,
+    children,
+    'data-testid': testId,
+    priority,
+    ...other
+  } = props
+  const [isLoadingState, setIsLoadingState] = useState(true)
+  const isLoading = isLoadingProp ?? isLoadingState
+  const { color, cornerRadius, motion } = useTheme()
+
+  const imageSource = !source
+    ? source
+    : typeof source === 'number'
+    ? source
+    : Array.isArray(source)
+    ? { uri: source[0].uri, priority }
+    : { uri: source.uri, priority }
```

Comment:
wonder if we can pick a better indentation rule in prettier that lets these ternaries nest in a more visually obvious way

Reply:
https://eslint.org/docs/latest/rules/indent#offsetternaryexpressions 👀